File: solutions/5976/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Sun Jan  9 16:03:22 2022
> Description:   
 ************************************************************************/
#include <algorithm> 
#include <numeric> 
#include <vector> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    bool checkValid(vector<vector<int>>& matrix) {
        const int n = matrix.size();
        for (int i = 0; i < n; ++i) {
            vector<int> recordRow(n, 0);
            vector<int> recordCol(n, 0);
            for (int j = 0; j < n; ++j) {
                recordRow[matrix[i][j]-1] = 1;
                recordCol[matrix[j][i]-1] = 1;
            }
            if (accumulate(recordRow.begin(), recordRow.end(), 0) != n) return false;
            if (accumulate(recordCol.begin(), recordCol.end(), 0) != n) return false;
        }
        return true;
    }
};

TEST(Test5976, SimpleTest) {
    vector<vector<int>> matrix{
        {
            1,2,3
        }, 
        {
            3,1,2
        }, 
        {
            2,3,1
        }
    };
    Solution s; 
    EXPECT_TRUE(s.checkValid(matrix));
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}

File: solutions/5976/main.py
from typing import List 

class Solution:
    def checkValid(self, matrix: List[List[int]]) -> bool:
        n = len(matrix)
        for i in range(n):
            recordRow = [0]*n
            recordCol = [0]*n
            for j in range(n):
                recordRow[matrix[i][j]-1] = 1
                recordCol[matrix[j][i]-1] = 1
            if sum(recordRow) != n or sum(recordCol) != n:
                return False
        return True

if __name__ == "__main__": 
    matrix = [[1,2,3],[3,1,2],[2,3,1]]
    s = Solution()
    assert s.checkValid(matrix)
